WireDomain/Sources/WireDomain/Repositories/FeatureConfigRepository.swift

```diff
@@ -0,0 +1,363 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+import Combine
+import WireAPI
+import WireDataModel
+
+/// Facilitates access to feature configs related domain objects.
+
+protocol FeatureConfigRepositoryProtocol {
+
+    /// Pulls feature configs from the server and stores them locally.
+    ///
+    /// - Returns: A list of `FeatureState`.
+
+    func pullFeatureConfigs() async throws -> [FeatureState]
+
+    /// Observes feature states.
+    ///
+    /// Each time `pullFeatureConfigs()` is called, a feature config is
+    /// stored locally and a new `FeatureState` value is produced by the publisher.
+    /// It allows the user to be notified of any feature changes over time.
+    ///
+    /// ⚠️ Use this method before calling `pullFeatureConfigs` to receive all emitted values.
+    ///
+    /// - Returns: A publisher of `FeatureState`.
+
+    func observeFeatureStates() -> AnyPublisher<FeatureState, Never>
+
+    /// Fetches a feature config locally.
+    ///
+    /// - Parameter name: The feature name to fetch the config for.
+    /// - Parameter type: The type of config to retrieve.
+    /// - Returns: A `LocalFeature` object with a status and a config (if any).
+
+    func fetchFeatureConfig<T: Decodable>(withName name: Feature.Name, type: T.Type) async throws -> LocalFeature<T>
+
+    /// Fetches a flag indicating whether the user should be notified of a given feature.
+    /// - Parameter name: The feature name.
+    /// - Returns: `true` if user should be notified.
+
+    func fetchNeedsToNotifyUser(forFeatureName name: Feature.Name) async throws -> Bool
+
+    /// Stores a flag indicating whether the user should be notified of a given feature.
+    /// - Parameter notifyUser: Whether the user should be notified for a given feature.
+    /// - Parameter name: The name of the feature to set the flag for.
+
+    func storeNeedsToNotifyUser(_ notifyUser: Bool, forFeatureName name: Feature.Name) async throws
+
+}
+
+final class FeatureConfigRepository: FeatureConfigRepositoryProtocol {
+
+    // MARK: - Properties
+
+    private let featureConfigsAPI: any FeatureConfigsAPI
+    private let context: NSManagedObjectContext
+    private let featureStateSubject = PassthroughSubject<FeatureState, Never>()
+
+    // MARK: - Object lifecycle
+
+    init(
+        featureConfigsAPI: any FeatureConfigsAPI,
+        context: NSManagedObjectContext
+    ) {
+        self.featureConfigsAPI = featureConfigsAPI
+        self.context = context
+    }
+
+    // MARK: - Public
+
+    func pullFeatureConfigs() async throws -> [FeatureState] {
```

Comment:
Do we still need to return something here? In other repositories we usually don't return anything from a pull, and when we want to fetch an object there is a separate fetch method which takes it from the local store.

Reply:
No, I forgot to remove it, fixed in last commit thx